db-es-audit/src/find_granules.py

```
@@ -0,0 +1,443 @@
+    ### need to create a new index (PARENT_COLLECTION_ID, REVISION_DATE, CONCEPT_ID, REVISION_ID, DELETED) on the granule tables.
+
+    try:
+        with db_connection.cursor() as curr:
+            curr.execute(search_query)
+            while True:
+                batch = curr.fetchmany(BATCH_SIZE)
+                if not batch:
+                    break
+                yield batch
+    except oracledb.Error as e:
+            raise Exception(f"Failed to execute query: {search_query}\nerror: {e}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)